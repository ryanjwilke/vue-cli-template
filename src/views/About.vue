<template>
  <div class="about">
    <h1>This is an about page</h1>

    <hr />

    <h2>Hello World Function</h2>
    <p>{{ helloWorld }}</p>

    <h2>Hello World Async Function</h2>
    <p>{{ helloWorldAsync }}</p>

    <h2>FaunaDB Function</h2>
    <p>{{ fauna }}</p>
  </div>
</template>

<script>
  import axios from 'axios'
  export default {
    name: 'about',
    beforeMount() {
      axios
        .get('/.netlify/functions/hello')
        .then((resp) => { this.helloWorld = resp.data.msg })
        .catch(() => { this.helloWorld = 'An error occurred.' })

      axios
        .get('/.netlify/functions/hello-async')
        .then((resp) => { this.helloWorldAsync = resp.data.msg })
        .catch(() => { this.helloWorldAsync = 'An error occurred.' })

      axios
        .get('/.netlify/functions/fauna')
        .then((resp) => { this.fauna = JSON.stringify(resp.data) })
        .catch(() => { this.fauna = 'An error occurred.' })
    },
    data() {
      return {
        fauna: 'Placeholder',
        helloWorld: 'Placeholder',
        helloWorldAsync: 'Placeholder',
      }
    }
  }
</script>
