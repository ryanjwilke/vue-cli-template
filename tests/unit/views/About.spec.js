import { shallowMount } from '@vue/test-utils'
import About from '@/views/About.vue'

describe('About', () => {
  const wrapper = shallowMount(About)

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders correctly with placeholder data', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it.skip('renders correctly with synca data', async () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
