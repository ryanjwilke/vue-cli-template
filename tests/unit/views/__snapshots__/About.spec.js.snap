// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly with placeholder data 1`] = `
<div
  class="about"
>
  <h1>
    This is an about page
  </h1>
   
  <hr />
   
  <h2>
    Hello World Function
  </h2>
   
  <p>
    Placeholder
  </p>
   
  <h2>
    FaunaDB Function
  </h2>
   
  <p>
    Placeholder
  </p>
</div>
`;

exports[`About renders correctly with synca data 1`] = `
<div
  class="about"
>
  <h1>
    This is an about page
  </h1>
   
  <hr />
   
  <h2>
    Hello World Function
  </h2>
   
  <p>
    Placeholder
  </p>
   
  <h2>
    FaunaDB Function
  </h2>
   
  <p>
    Placeholder
  </p>
</div>
`;
